pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -115,10 +116,21 @@ func (c *Controller) Run(numWorkers int, stopCh <-chan struct{}, controllerManag
 		},
 		// this could result in a large amount of updates, but we cap the number of possible networks to avoid those issues
 		UpdateFunc: func(old, new interface{}) {
-			network := new.(*networkv1.Network)
-			if network.Spec.ParametersRef != nil && network.Spec.ParametersRef.Kind == gnpKind {
-				c.queue.Add(network.Spec.ParametersRef.Name)
+			newNetwork := new.(*networkv1.Network)
+			if newNetwork.Spec.ParametersRef != nil && newNetwork.Spec.ParametersRef.Kind == gnpKind {
+				c.queue.Add(newNetwork.Spec.ParametersRef.Name)
+			}
+
+			// we need to check the old network to see if we are no longer referencing the same GNP
+			// this important so we can delete a GNP waiting for a Network to no longer be inuse.
+			oldNetwork := old.(*networkv1.Network)
+			bothNetworksHaveParams := oldNetwork.Spec.ParametersRef != nil && newNetwork.Spec.ParametersRef != nil
+			paramsRefKindChangedOffGNP := bothNetworksHaveParams && oldNetwork.Spec.ParametersRef.Kind == gnpKind && newNetwork.Spec.ParametersRef.Kind != gnpKind
+			paramsRefNameChanged := bothNetworksHaveParams && oldNetwork.Spec.ParametersRef.Name != newNetwork.Spec.ParametersRef.Name
```

Comment:
In theory you will process this even if both networks were NOT referring to a GNP kind and the param name changed.

I think it is much simpler to do this in this way:

```
if oldNetwork.Spec.ParametersRef != nil && oldNetwork.Spec.ParametersRef.Kind == gnpKind {
    if newNetwork.Spec.PrametersRef == nil || newNetwork.Spec.ParamtersRef != gnpKind || newNetwork.Spec.ParametersRef.Name != newNetwork.Spec.ParametersRef.Name {
        c.queue.Add(oldNetwork.Spec.ParametersRef.Name)
    }
}
```

Reply:
thanks, I think this is why the test was failing